### PR TITLE
Cleanup deployment docs and add azure

### PR DESCRIPTION
**Pull request by derrickmehaffy**
#### Description of what you did:

**This is a WIP PR**

- [x] Create new Deployment docs structure
- [x] Move existing deployment docs into their own files for easier management
- [ ] ~~Fix/Clean up any existing deployment docs that aren't right~~
- [x] Move any reusable documentation to it's own file and create references
- [x] Add Azure documentation

### CODE REVIEW

**Comment by derrickmehaffy** [comment]
@lauriejim @alexandrebodin @soupette 

If any of you can look over my changes (I think there is a bit from other PRs that is not here) but it would be best to just get what I have done so far merged in so that other PRs are not based on the old structure.

I will do another PR for the new proxy config changes as well as the split stuff that I discussed with @lauriejim 

Eventually I would prefer to have small little deployment "guides" that each new environment can just link to such as webhook auto-deploying and what not.

We duplicate so much for example between AWS and DigitalOcean and the code has only slight differences, as we continue to add more this is going to get confusing and harder to manage.

**Comment by alexandrebodin** [comment]
I think your guides will have to change with this PR https://github.com/strapi/strapi/pull/5747. 
Once @lauriejim reviews this PR and merges it can you please update those guides to be. work with the .env approache + exploiting Paas platforms with env managment better.

File: docs/3.0.0-beta.x/deployment/proxy.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
+# Proxying
```
Comment (lauriejim) [inline docs/3.0.0-beta.x/deployment/proxy.md:1]: I think this part has to be in the `docs/3.0.0-beta.x/getting-started/deployment.md` file.
This is not a provider but just configurations for the server.

File: docs/3.0.0-beta.x/getting-started/deployment.md [hunk 2]
```diff
@@ -0,0 +1,190 @@
+# Deployment
+
+Strapi gives you many possible deployment options for your project or application. Strapi can be deployed on traditional hosting servers or services such as Heroku, AWS, Azure and others. The following documentation covers how to develop locally with Strapi and deploy Strapi with various hosting options.
+
+::: tip
+Deploying **databases** along with Strapi is covered in the [Databases Guide](../guides/databases.md).
+:::
+
+### Documented Options
```
Comment (lauriejim) [inline docs/3.0.0-beta.x/getting-started/deployment.md:9]: I think we can rename this section with a `##` and named it something like `Hosting provider guides`
Just to mention that the followings are deployment guides for a specific platform.

File: docs/3.0.0-beta.x/getting-started/deployment.md [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+    </template>
+		<template #title>Heroku</template>
+		<template #description>
+			Step by step guide for deploying on Heroku
+		</template>
+	</InstallLink>
+</div>
+
+<div>
+	<InstallLink link="../deployment/proxy">
```
Comment (lauriejim) [inline docs/3.0.0-beta.x/getting-started/deployment.md:62]: Remove this proxy documentation here.

File: docs/3.0.0-beta.x/getting-started/deployment.md [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+    <svg xmlns="http://www.w3.org/2000/svg" viewBox="-35.5 26 32 32" width="64" height="64"><path d="M-33.442 42.023v-7.637a.68.68 0 0 1 .385-.651l13.173-7.608c.237-.148.503-.178.74-.03l13.232 7.637a.71.71 0 0 1 .355.651V49.63a.71.71 0 0 1-.355.651l-11.367 6.57a56.27 56.27 0 0 1-1.806 1.036c-.266.148-.533.148-.8 0l-13.202-7.608c-.237-.148-.355-.326-.355-.622v-7.637z" fill="#fff"/><path d="M-24.118 39.18v8.9c0 1.006-.8 1.894-1.865 1.865-.65-.03-1.154-.296-1.5-.858-.178-.266-.237-.562-.237-.888V35.836c0-.83.503-1.42 1.154-1.687s1.302-.207 1.954 0c.622.178 1.095.562 1.5 1.036l7.874 9.443c.03.03.06.09.118.148v-9c0-.947.65-1.687 1.57-1.776 1.154-.148 1.924.68 2.042 1.54v12.6c0 .7-.326 1.214-.918 1.54-.444.237-.918.296-1.42.266a3.23 3.23 0 0 1-1.954-.829c-.296-.266-.503-.592-.77-.888l-7.49-8.97c0-.03-.03-.06-.06-.09z" fill="#3498DB"/></svg>
+    </template>
+		<template #title>Proxy</template>
+		<template #description>
+			Overview of various Proxy options and sample configurations
+		</template>
+	</InstallLink>
+</div>
+
+### Configuration
```
Comment (lauriejim) [inline docs/3.0.0-beta.x/getting-started/deployment.md:73]: Transform this in `##` and rename with `Application configuration` just to precise that is for the Strapi app.

File: docs/3.0.0-beta.x/getting-started/deployment.md [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+:::
+
+If you need a server.js file to be able to run `node server.js` instead of `npm run start` then create a `./server.js` file as follows:
+
+```js
+const strapi = require('strapi');
+
+strapi(/* {...} */).start();
+```
+
```
Comment (lauriejim) [inline docs/3.0.0-beta.x/getting-started/deployment.md:187]: Add the proxy configuration here with a `##`

**Comment by lauriejim** [comment]
A warning message for us, we will have to manage the permanent redirection for the previous URL to the new one.

**Comment by derrickmehaffy** [comment]
@lauriejim Ready for re-review

**Pull request merged**